aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -1385,7 +1385,7 @@ Tensor from_file(std::string filename, c10::optional<bool> shared, c10::optional
         c10::StorageImpl::use_byte_size_t(),
         size_bytes,
         THMapAllocator::makeDataPtr(
-            filename.c_str(), flags, size_bytes, nullptr),
+            std::string(filename).c_str(), flags, size_bytes, nullptr),
```

Comment:
Since `filename` used to be `std::string`, it was already being copied. So in this case, it shouldn't really make a difference

Reply:
This would be more obviously correct if you went and updated the signature on `THMapAllocator::makeDataPtr`. It takes a `const char*` and then copies it into a `std::string`. There's no reason to do this: make it take a `std::string` by value and move it in. That'll solve all your problems.